homeassistant/components/garmin_connect/__init__.py

```diff
@@ -93,16 +93,43 @@ def __init__(self, hass, client):
         today = date.today()
 
         try:
-            self.data = self.client.get_stats(today.isoformat())
+            activity_data = self.client.get_stats(today.isoformat())
         except (
             GarminConnectAuthenticationError,
             GarminConnectTooManyRequestsError,
+            GarminConnectConnectionError,
         ) as err:
-            _LOGGER.error("Error occurred during Garmin Connect get stats: %s", err)
+            _LOGGER.error(
+                "Error occurred during Garmin Connect get activity data: %s", err
+            )
             return
-        except (GarminConnectConnectionError) as err:
-            _LOGGER.error("Error occurred during Garmin Connect get stats: %s", err)
+        except Exception:  # pylint: disable=broad-except
+            _LOGGER.error(
```

Comment:
We should log this with exception since it's unknown and not expected.

Reply:
You mean split these up with their own -more descriptive- error message?
        except (
            GarminConnectAuthenticationError,
            GarminConnectTooManyRequestsError,
            GarminConnectConnectionError,
        ) as err:

Reply:
No, just change to use `_LOGGER.exception` instead of `_LOGGER.error`.

Reply:
I have change the garminconnect python module to do both requests and concat of the dicts, and I have replaced only the unkown logger.error entries with logger.exception. Hope I got it right.